rascal-lsp/src/main/rascal/demo/lang/pico/NewLanguageServer.rsc

```diff
@@ -0,0 +1,187 @@
+@license{
+Copyright (c) 2018-2023, NWO-I CWI and Swat.engineering
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+
+1. Redistributions of source code must retain the above copyright notice,
+this list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright notice,
+this list of conditions and the following disclaimer in the documentation
+and/or other materials provided with the distribution.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+POSSIBILITY OF SUCH DAMAGE.
+}
+module demo::lang::pico::NewLanguageServer
+
+import util::LanguageServer;
+import util::IDEServices;
+import ParseTree;
+import util::Reflective;
+import lang::pico::\syntax::Main;
+
+@synopsis{Provides each contribution (IDE feature) as a callback element of the set of LanguageServices.}
+set[LanguageService] picoLanguageContributor() = {
+    parsing(parser(#start[Program])),
+    documentSymbol(picoOutliner),
+    codeLense(picoLenses),
+    execution(picoCommands),
+    inlayHint(picoHinter),
+    definition(lookupDef),
+    codeAction(picoActions)
```

Comment:
I imagine Rascal users may look at this module for inspiration when they want to implement their own language server (to the extent of just copy-pasting the whole thing and find-replacing `pico` with `myLanguage`). If so, then I suppose the code in this module should reflect what we believe is good coding style for language servers. Long story short, maybe we should rename the arguments to be more consistent with the names of the constructors.

Reply:
Agreed, we should rename them and use some of the old words in the synopsis and descriptions. E.g. documentSymbol should mention the word outline. 

This stuff is also still inspiration for people who are porting rascal eclipse plugins.